workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java

```diff
@@ -359,4 +360,9 @@ private String validateWorkflow(String workflowName, WorkFlow workFlow) {
 		return null;
 	}
 
+	@PreDestroy
+	public void gracefulShutdown() {
+		log.info(" >>>>>>>>> Shutting down the workflow service <<<<<<<<<<<");
```

Comment:
@lshannon @RichardW98 @anludke which logic would we like to have in this context?
Right before the application ends, there are no active requests, but there may be running executions.

Reply:
yes we need to wait for running tasks and their persistence to complete before shutdown, and prevent upcoming workflow/task runs